crates/libs/bindgen/src/rust/winrt_methods.rs

```diff
@@ -47,38 +60,68 @@ pub fn writer(
         }
     };
 
+    let vcall = quote! { (windows_core::Interface::vtable(this).#vname)(windows_core::Interface::as_raw(this), #args #return_arg) };
+
     let vcall = match &signature.return_type {
         metadata::Type::Void => {
-            quote! {
-                (windows_core::Interface::vtable(this).#vname)(windows_core::Interface::as_raw(this), #args).ok()
+            if noexcept {
+                quote! {
+                    let hresult__ = #vcall;
+                    debug_assert!(hresult__.0 == 0);
```

Comment:
Suggest:

```
debug_assert_eq!(hresult__.0, 0, "Method marked `noexcept` returned a failure code");
```

Reply:
I'm reluctant to add this since it can be repeated oodles of times in generated code. Let's wait and see whether this assertion raises any questions and reconsider. As it is, both sides of the contract will almost always be generated by tools - `cppwinrt` or `windows-bindgen` - and it would be unusual for somebody to write this ABI code by hand.